### PR TITLE
Expose erlcloud_cloudwatch_logs:start_query/stop_query/get_query_results

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
~~I'm pushing this as a draft first (it's missing tests) to try and get early feedback.~~

~~Stuff I did:~~ This pull request:
- exposes functions `erlcloud_cloudwatch_logs:get_query_results/2,3`
- exposes functions `erlcloud_cloudwatch_logs:start_query/4,5,6`
- exposes functions `erlcloud_cloudwatch_logs:stop_query/1,2`
- exposes type `erlcloud_cloudwatch_logs:query_status/0`
- exposes type `erlcloud_cloudwatch_logs:query_results/0`
- has input/output as `string()` always, for consistency
- has output as `map()`-based (optional for `:get_query_results`), and transformed in a way that keeps a reference to the original (check doc.) but is more "native" to Erlang consumers (e.g. `atom()` and `map()`)

### CODE REVIEW

**Comment by paulo-ferraz-oliveira** [comment]
FWIW: `[{<<"rejectedLogEventsInfo">>,[{<<"tooOldLogEventEndIndex">>,1}]}]` is returned by AWS, for `:put_logs_events/5`, but hidden by the `erlcloud` API.

File: src/erlcloud_cloudwatch_logs.erl
Comment (nalundgaard) [inline src/erlcloud_cloudwatch_logs.erl:480]: Would it be better to pass `Config` here explicitly?

**Review by nalundgaard — CHANGES_REQUESTED** [review]
I like the map result format pretty well, but I'm not a huge fan of introducing it into an API that uses proplist-format JSX terms for all of the other API endpoints. Would it be better to pursue something more like #682, where there's an `{out, map}` option? I guess this would take away the nice atoms, and I guess on the input side #511 already kinda made this API somewhat inconsistent. I'm happy to approve if you prefer it as you have it, given the state of the module, just want to confirm.

**Comment by paulo-ferraz-oliveira** [comment]
> I like the map result format pretty well, but I'm not a huge fan of introducing it into an API that uses proplist-format JSX terms for all of the other API endpoints. Would it be better to pursue something more like #682, where there's an `{out, map}` option? I guess this would take away the nice atoms, and I guess on the input side #511 already kinda made this API somewhat inconsistent. I'm happy to approve if you prefer it as you have it, given the state of the module, just want to confirm.

I don't mind changing this to support `{out, map}`, but would prefer to not have to change the non-map output, if that's OK.

File: src/erlcloud_cloudwatch_logs.erl
Comment (paulo-ferraz-oliveira) [reply src/erlcloud_cloudwatch_logs.erl]: Sure: 87868d5.

**Comment by paulo-ferraz-oliveira** [comment]
> I don't mind changing this to support `{out, map}`, but would prefer to not have to change the non-map output, if that's OK.

5e31836

**Comment by paulo-ferraz-oliveira** [comment]
@nalundgaard, is there something you feel is blocking merging?

(if you do merge, I'd also like to request a release for inclusion as a tagged API, if that's OK)

File: src/erlcloud_cloudwatch_logs.erl
Comment (nalundgaard) [inline src/erlcloud_cloudwatch_logs.erl:497]: Should this type spec be adjusted to account for the non-`query_results()` `{ok, ...}` form?

**Comment by nalundgaard** [comment]
Sure, sorry I forgot to come back to this earlier in the week. I had one concern about the type spec after the out opt, otherwise LGTM, and will make a tag for you when it's merged for sure. Thanks!

File: src/erlcloud_cloudwatch_logs.erl
Comment (paulo-ferraz-oliveira) [reply src/erlcloud_cloudwatch_logs.erl:497]: You are right. I was expecting the `dialyzer` to have a say, here, but it didn't :-(

**Review by nalundgaard — APPROVED** [review]

**Comment by paulo-ferraz-oliveira** [comment]
Cool, thanks.

**Comment by nalundgaard** [comment]
Thank you! I have made a release and published it to Hex. I'm kicking myself for not using squash and merge, but with time and therapy (and newer git history in the repo), I'm confident that the healing will take place.

Have a good one!

**Comment by paulo-ferraz-oliveira** [comment]
😄 some people prefer a more detailed history (because there was a thought process behind the change), some people prefer a "cleaner" history, but one that shows only an increment. I have no preference for the lib.s I contribute to, and always expect the maintainers to let me know how they'd prefer it (e.g. I've had pull requests accepted into the same repo with request to merge + squash and without it, go figure 😄). I do prefer to have the merge commit, though, so I understand where the change comes from; some people don't...

**Pull request merged**
